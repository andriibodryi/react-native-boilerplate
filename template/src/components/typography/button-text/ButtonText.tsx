import * as React from 'react';

import { BasicText } from '../basic-text';
import { useBasicTextStyles } from './button-text.presets';
import { ButtonTextTypes } from './button-text.types';

export const ButtonText: React.FC<ButtonTextTypes> = ({
  preset,
  style,
  ...rest
}) => {
  const styles = useBasicTextStyles();

  return <BasicText style={[styles[preset], style]} {...rest} />;
};

export default ButtonText;
