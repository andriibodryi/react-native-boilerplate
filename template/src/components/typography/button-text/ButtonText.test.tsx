import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { ButtonText } from './ButtonText';

describe('ButtonText', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<ButtonText preset="outlineButton">basic text</ButtonText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
