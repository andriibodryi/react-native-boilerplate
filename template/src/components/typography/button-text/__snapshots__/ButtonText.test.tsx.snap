// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonText should render correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4D4D4D",
        "fontFamily": "Aspira-Regular",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      Array [
        Object {
          "fontFamily": "Aspira-Medium",
          "fontSize": 22,
          "fontWeight": "500",
          "lineHeight": 26,
        },
        Object {
          "color": "#4D4D4D",
        },
        undefined,
      ],
    ]
  }
>
  basic text
</Text>
`;
